software/gui/gui.sh

```diff
@@ -153,7 +156,23 @@ if [ "$1" == "--build" ]; then
     j_opt="-j"
   fi
 
-  pushd build && qmake $config_opt .. && make $j_opt && popd
+  pushd build && qmake $config_opt .. && bear make $j_opt && popd
+
+  cd build
+
+  if [ "$3" != "--no-checks" ]; then
+    set +e
+    set +o pipefail
+  fi
+
+  cppcheck --project=compile_commands.json -i../src/third_party .
```

Comment:
ok, I guess this means the one above is a duplicate?

Reply:
This second cppcheck uses the compile database to check the build files whereas the first one checks everything before building. I suppose then we need a clang-tidy check before building too for consistency.

Reply:
Nevermind. Looking into the CI job's output, it's checking those non-build files too.